### PR TITLE
Trigger notification for events

### PR DESCRIPTION
**Pull request by jamil314**
## Description

Clearly describe what has been changed. Include relevant context or background.
Explain how the issue was fixed (if applicable) and the root cause.

Link this pull request to the GitHub issue (and optionally name the branch `ocrvs-<issue #>`)

## Checklist

- [ ] I have linked the correct Github issue under "Development"
- [ ] I have tested the changes locally, and written appropriate tests
- [ ] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [ ] I have updated the GitHub issue status accordingly

### CODE REVIEW

**Review by cibelius — COMMENTED** [review]
First batch of PR comments. I think that `informationNotification.ts` needs a larger refactor, e.g. by using early returns and such.

File: src/api/notification/handler.ts [hunk 1]
```diff
@@ -101,33 +106,50 @@ function generateFailureLog({
...
-    applicationConfig.USER_NOTIFICATION_DELIVERY_METHOD
+export async function notify({
+  event,
+  variable,
+  recipient,
+  deliveryMethod
+}: (UserEventVariablePair | InformantEventVariablePair) & {
+  recipient: Recipient
+  deliveryMethod: string
+}) {
```
[the remark this answers is not in the document]
Comment (jamil314) [reply src/api/notification/handler.ts]: That is better. Thanks.

**Review by cibelius — APPROVED** [review]
Nice 🙌
